addons/account/models/account_move.py

```diff
@@ -2988,6 +2992,43 @@ def _compute_balance(self):
         for line in self:
             line.balance = line.debit - line.credit
 
+    @api.model
+    def search_read(self, domain=None, fields=None, offset=0, limit=None, order=None):
+        def to_tuple(t):
+            return tuple(map(to_tuple, t)) if isinstance(t, (list, tuple)) else t
+        # Add the domain and order by in order to compute the cumulated balance in _compute_cumulated_balance
+        return super(AccountMoveLine, self.with_context(domain_cumulated_balance=to_tuple(domain or []), order_cumulated_balance=order or self._order)).search_read(domain, fields, offset, limit, order)
+
+    @api.depends_context('order_cumulated_balance', 'domain_cumulated_balance')
+    def _compute_cumulated_balance(self):
+        # reverse the order by
+        order_list = [x.split() for x in (self.env.context.get('order_cumulated_balance')).split(',')]
+        for order in order_list:
+            order[0] = 'account_move_line.' + order[0]
+            if len(order) == 1:
+                order += ['DESC']
+            elif order[1].upper() == 'ASC':
+                order[1] = 'DESC'
+            else:
+                order[1] = "ASC"
+            assert len(order) == 2  # This should prevent sql injections for untrusted context
```

Comment:
mhm It will, but I find it a little tedious. Don't you prefer using regexp for that?

Reply:
okay then